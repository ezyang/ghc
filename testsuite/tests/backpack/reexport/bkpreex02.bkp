unit p where
    signature T where
        data T
    signature H where
        import T
        f :: T -> T
unit timpl where
    module TImpl where
        data T = T
unit q where
    include timpl
    signature T(T) where
        import TImpl
    include p
    module A where
        import H
        import TImpl
        x = f T
unit r-impl where
    include timpl
    module H where
        import TImpl
        f T = T
    module T(T) where
        import TImpl
unit r where
    include r-impl
    include q
