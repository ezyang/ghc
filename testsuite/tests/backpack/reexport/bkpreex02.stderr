[1 of 5] Processing p
  [1 of 2] Compiling T[sig]           ( p/T.hsig, nothing )
  [2 of 2] Compiling H[sig]           ( p/H.hsig, nothing )
[2 of 5] Processing timpl
  Instantiating timpl
  [1 of 1] Compiling TImpl            ( timpl/TImpl.hs, nothing )
[3 of 5] Processing q
  Checking timpl
  [1 of 3] Compiling T[sig]           ( q/T.hsig, nothing )
  [2 of 3] Compiling H[sig]           ( q/H.hsig, nothing )
  [3 of 3] Compiling A                ( q/A.hs, nothing )

bkpreex02.bkp:18:15: error:
    • Couldn't match expected type ‘hole:T.T’
                  with actual type ‘T’
      NB: ‘T’ is defined at bkpreex02.bkp:9:9-18
          ‘hole:T.T’ is defined at bkpreex02.bkp:3:9-14
    • In the first argument of ‘f’, namely ‘T’
      In the expression: f T
      In an equation for ‘x’: x = f T
